cc_plugin_ncei/ncei_base.py

```diff
@@ -0,0 +1,689 @@
+                enumerated_flag_variable:flag_meanings = "" ; //............. REQUIRED    - Provide a space-separated list of meanings corresponding to each of the flag_values
+                enumerated_flag_variable:references = "" ; //................ RECOMMENDED - Published or web-based references that describe the data or methods used to produce it.
+                enumerated_flag_variable:comment = "" ; //................... RECOMMENDED - Add useful, additional information here.
+                '''
+        #Check the qaqc variables to ensure they are good
+        results = []
+        msgs = []
+        std_name_list = [getattr(dataset.variables[var], 'standard_name', None) for var in dataset.variables]
+        for name in dataset.variables:
+            if hasattr(dataset.variables[name],'flag_meanings'):
+                #Check Standard Name
+                std_check = 0
+                msgs = []
+
+                std_name = getattr(dataset.variables[name], 'standard_name', 'thereis noname')
+                std_name_split = std_name.split(' ')
+                if std_name_split[0] in std_name_list:
+                    std_check = std_check + 1
+                else:
+                    msgs.append('Standard Name does not reference a variable in the dataset.')
+                if std_name_split[1] == 'status_flag':
+                    std_check = std_check + 1
+                else:
+                    msgs.append('Standard Name does not end in "status_flag"')
+                results.append(Result(BaseCheck.HIGH, (std_check,2), (name,'standard_name'), msgs))
+
+                #Check Flag Mask / Flag Values
+                msgs = []
+                mask_val_check = False
+                test_name = 'flag_masks_values'
+                if 'bool' in str(dataset.variables[name].dtype):
+                    mask_val_check = hasattr(dataset.variables[name],'flag_masks')
+                    test_name = 'flag_masks'
+                    mask_length = len(getattr(dataset.variables[name], 'flag_masks', None)) #Used in length check
+                elif 'int' in str(dataset.variables[name].dtype):
+                    mask_val_check = hasattr(dataset.variables[name], 'flag_values')
+                    test_name = 'flag_values'
+                    mask_length = len(getattr(dataset.variables[name], 'flag_values', None)) #Used in length check
+                else:
+                    mask_val_check = False
+                    msgs = ['flag_masks or flag_values are not present']
+                    mask_length = 0
+                results.append(Result(BaseCheck.HIGH, mask_val_check, (name,test_name), msgs))
+
+                #Check Flag Meaning
+                msgs = []
+                if hasattr(dataset.variables[name],'flag_meanings'):
+                    meanings_check = True
+                    meaning_length = len(getattr(dataset.variables[name],'flag_meanings',None).split(','))
```

Comment:
I recognize that NODC template 1.1 says comma separated but that directly competes with practices and CF:

http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html#flag-variable-flag-values-ex

And in the newer template they address this

```
enumerated_flag_variable:flag_meanings = "" ; //............. REQUIRED    - Provide a space-separated list of meanings corresponding to each of the flag_values
```